### ISSUE TITLE
Block size unexpected, causing file read throw I/O error

### ISSUE THREAD

**Issue by Nizifan** [Alluxio/alluxio#15288]
**Alluxio Version:**
2.7.2

**Describe the bug**
User fail to read multiple files consistently:
<img width="743" alt="wecom-temp-97f2aacf5ba9658c7675207fd2aea176" src="https://user-images.githubusercontent.com/11472213/162660843-78a521e8-f3d4-484f-9d34-87ce28243267.png">

If using Alluxio command to list/copyToLocal/free, receiving below error:

[root@jssz-admpi-147 alluxio-2.7.2]# alluxio fs ls /379009661_nb2-1-74/ | grep 20205
         833320       PERSISTED 08-16-2021 20:09:52:000 100% /379009661_nb2-1-74/020205.png
            317       PERSISTED 08-16-2021 20:09:52:000   0% /379009661_nb2-1-74/020205.json

[root@jssz-admpi-147 alluxio-2.7.2]# alluxio fs copyToLocal /379009661_nb2-1-74/020205.png .
Block 48551283392512 is expected to be 833320 bytes, but only 0 bytes are available. Please ensure its metadata is consistent between Alluxio and UFS.

[root@jssz-admpi-147 alluxio-2.7.2]# alluxio fs free /379009661_nb2-1-74/020205.png
java.util.concurrent.TimeoutException: Timed out waiting for file to be freed. Another user may be loading it. options: WaitForOptions{interval=1000, timeout=10000} last value: false

The next time try copyToLocal, it succeed.

If we try get log from worker, didn't find anything related to "48551283392512" and "Error"
root@jssz-coeus-alluxio-01:~# k logs qa-frames-worker-4vwpr alluxio-worker | grep rror
root@jssz-coeus-alluxio-01:~# k logs qa-frames-worker-4vwpr alluxio-worker | grep 48551283392512
Both return nothing.

And user reported more files hitting this issue, using stat can see below information

[root@jssz-admpi-147 alluxio-2.7.2]# alluxio fs stat /248251145_nb2-1-64/023299.png  
/248251145_nb2-1-64/023299.png is a file path.
URIStatus{info=FileInfo{fileId=10920591884287, name=023299.png, path=/248251145_nb2-1-64/023299.png, ufsPath=s3://cv_data/qichang/qa/frames/248251145_nb2-1-64/023299.png, length=1064509, blockSizeBytes=16777216, creationTimeMs=1649229190341, completed=true, folder=false, pinned=false, pinnedlocation=[], cacheable=true, persisted=true, blockIds=[10920575107072], inMemoryPercentage=0, lastModificationTimesMs=1629259544000, ttl=-1, lastAccessTimesMs=1649363203245, ttlAction=FREE, owner=root, group=root, mode=448, persistenceState=PERSISTED, mountPoint=false, replicationMax=1, replicationMin=0, fileBlockInfos=[FileBlockInfo{blockInfo=BlockInfo{id=10920575107072, length=1064509, locations=[BlockLocation{workerId=2768092020693031836, address=WorkerNetAddress{host=10.72.70.19, containerHost=, rpcPort=20200, dataPort=20200, webPort=20201, domainSocketPath=, tieredIdentity=TieredIdentity(node=10.72.70.19, rack=null)}, tierAlias=SSD, mediumType=SSD}]}, offset=0, ufsLocations=[]}], mountId=1, inAlluxioPercentage=100, ufsFingerprint=TYPE|FILE UFS|s3 OWNER|_ GROUP|_ MODE|448 CONTENT_HASH|a8c1246ad132fe54567ac7d961273a2b , acl=user::rwx,group::---,other::---, defaultAcl=}, cacheContext=null}
Containing the following blocks: 
BlockInfo{id=10920575107072, length=1064509, locations=[BlockLocation{workerId=2768092020693031836, address=WorkerNetAddress{host=10.72.70.19, containerHost=, rpcPort=20200, dataPort=20200, webPort=20201, domainSocketPath=, tieredIdentity=TieredIdentity(node=10.72.70.19, rack=null)}, tierAlias=SSD, mediumType=SSD}]}

**Comment by LuQQiu** [comment]
Occur 
```
Block 48551283392512 is expected to be 833320 bytes, but only 0 bytes are available. Please ensure its metadata is consistent between Alluxio and UFS.
```
@beinan looks like a similar issue happens. 
What could be the possible reasons that a block is broken?
- have blocks left by the previous Alluxio cluster, didn't fully clean up when launching the new alluxio cluster, and have the same block id in the new cluster.
- block itself broke after successfully written (unlikely?)
-

**Comment by HelloHorizon** [comment]
FYI @yuzhu

**Comment by beinan** [comment]
> Occur
> 
> ```
> Block 48551283392512 is expected to be 833320 bytes, but only 0 bytes are available. Please ensure its metadata is consistent between Alluxio and UFS.
> ```
> 
> @beinan looks like a similar issue happens. What could be the possible reasons that a block is broken?
> 
> * have blocks left by the previous Alluxio cluster, didn't fully clean up when launching the new alluxio cluster, and have the same block id in the new cluster.
> * block itself broke after successfully written (unlikely?)

Yes, @yuzhu and I have seen the same symptom when rolling back the journal.   As you said, there might be block files that have not been cleaned up. I'm still thinking to keep a last modified time in the metadata of each block will help alluxio kickout the staled or orphan blocks.

**Comment by LuQQiu** [comment]
From @Nizifan, When cache is incorrect, can we delete the block and read from UFS to get the correct one

**Comment by LuQQiu** [comment]
@beinan I discussed with @Nizifan  that 
in Fluid with 0.7, worker is launch with statefulset, which means that the worker can be launched on random machine.
e.g. The kubernetes cluster has 7 nodes, we want 5 workers, the workers can be launched on one of the machine. 

The worker may be randomly restarted and relaunch on other machines, but we want to reuse the local storage if any and if master has not been formatted.

e.g. 
Master start on Node1
WorkerA starts on Node2
WorkerA write some blocks in Node2.
WorkerA stops.
after a while
WorkerB starts on Node2, workerB wants to reuse WorkerA's data.

In this case, we may not want to record the worker_format_time, only <cluster_id> or say master_format_time / master_clean_start_time is needed.
